acquia/mc-cs-plugin-custom-objects
# MAUT-7885 : Block modification of custom object unique value if custom object has data

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | No
| New feature/enhancement? (use the a.x branch)      | Yes
| Deprecations?                          | No
| BC breaks? (use the c.x branch)        | No
| Automated tests included?              | Yes 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | https://backlog.acquia.com/browse/MAUT-7885 



#### Description:

Block edit operation of custom object if customer tries to  modify existing unique identifier for a custom field.

1) If data doesn't exists for Custom Object then allow modification of Unique Identifier value of Custom Fields.
2) If data exists for Custom Object then do not allow modify Unique Identifier value of Custom Fields
3) We can add new custom fields with unique identifier value for existing Custom Objects.

Note: If they want modify exiting unique value for custom object then they can create new custom object and all import custom items

**Status code for API**: 400



#### Steps to test this PR:


1. Open this PR for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Add Custom object with few fields.
3. Add some custom items.
4. Edit custom object field and change the "isUniqueIdentifier" field value and submit.
5. It should return the error message "System cannot modify unique identifier for the custom object which is in use".
6. And while adding new custom field with unique identifier to existing custom objects it should save.
7. Also verify 400 status code for Update request of Custom Fields API.
